File: SEGUNDO/PDOO/PRACTICAS/PRACTICA2/ruby/EnemyStarShip.rb
#encoding:utf-8

require_relative 'EnemyToUI.rb'
require_relative 'Loot.rb'
require_relative 'Damage.rb'
require_relative 'ShotResult.rb'

module Deepspace
    class EnemyStarShip
        def initialize(n, a, s, l, d)
            @name = n
            @ammoPower = a
            @shieldPower = s
            @loot = Loot.new(l.nSupplies, l.nWeapons, l.nShields, l.nHangars, l.nMedals)
            @damage = d.copy
        end

        def self.newCopy (e)
            new(e.name, e.ammoPower, e.shieldPower, e.loot, e.damage)
        end

        def getUIversion
            EnemyToUI.new(self)
        end

        public
        def fire
            @ammoPower
        end

        def ammoPower
            @ammoPower
        end

        def damage
            @damage
        end

        def loot 
            @loot
        end

        def name 
            @name
        end

        def shieldPower 
            @shieldPower
        end

        def protection
            @shieldPower
        end

        def receiveShot (shot)
            if (protection >= shot)
                ShotResult::RESIST
            else
                ShotResult::DONOTRESIST
            end
        end

        def to_s
            getUIversion.to_s
        end
    end
end

File: SEGUNDO/PDOO/PRACTICAS/PRACTICA2/java/deepSpace/src/deepspace/EnemyStarShip.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author joarpe
 */
public class EnemyStarShip {
    private float ammoPower;
    private String name;
    private float shieldPower;
    private Loot loot;
    private Damage damage;
    
    EnemyStarShip (String n, float a, float s, Loot l, Damage d) {
        ammoPower = a;
        name = n;
        shieldPower = s;
        loot = new Loot (l.getNSupplies(), l.getNWeapons(), l.getNShields(), l.getNHangars(), l.getNMedals());
        damage = d.copy();
    }
            
    EnemyStarShip (EnemyStarShip e) {
        ammoPower = e.getAmmoPower();
        name = e.getName();
        shieldPower = e.getShieldPower();
        loot = new Loot (e.getLoot().getNSupplies(), e.getLoot().getNWeapons(), e.getLoot().getNShields(), e.getLoot().getNHangars(), e.getLoot().getNMedals());
        damage = e.getDamage().copy();
    }
    
    EnemyToUI getUIversion () {
        return (new EnemyToUI(this));
    }
    
    public float fire () {
        return ammoPower;
    }
    
    public float getAmmoPower () {
        return ammoPower;
    }
    
    public Damage getDamage () {
        return damage;
    }
    
    public Loot getLoot () {
        return loot;
    }
    
    public String getName () {
        return name;
    }
    
    public float getShieldPower () {
        return shieldPower;
    }
    
    public float protection () {
        return shieldPower;
    }
    
    public ShotResult receiveShot (float shot) {
        if (shieldPower > shot) {
            return ShotResult.RESIST;
        }
        
        else {
            return ShotResult.DONOTRESIST;
        }
    }
    
    public String toString () {
        return getUIversion().toString();
    }
}
